presto-verifier/src/main/java/com/facebook/presto/verifier/rewrite/QueryRewriter.java

```diff
@@ -424,9 +426,19 @@ private Query rewriteNonStorableColumns(Query query, ResultSetMetaData metadata)
         checkState(selectItems.size() == columnTypes.size(), "SelectItem count (%s) mismatches column count (%s)", selectItems.size(), columnTypes.size());
         for (int i = 0; i < selectItems.size(); i++) {
             SingleColumn singleColumn = (SingleColumn) selectItems.get(i);
-            Optional<Type> columnTypeRewrite = getColumnTypeRewrite(columnTypes.get(i));
+            Type columnType = columnTypes.get(i);
+            Optional<Type> columnTypeRewrite = getColumnTypeRewrite(columnType);
             if (columnTypeRewrite.isPresent()) {
-                newItems.add(new SingleColumn(new Cast(singleColumn.getExpression(), columnTypeRewrite.get().getTypeSignature().toString()), singleColumn.getAlias()));
+                Expression expression = singleColumn.getExpression();
+                if (columnType.equals(GEOMETRY)) {
+                    // Geometry type not a Hive writable type, thus it should be converted to VARCHAR type in WKT
+                    // format using the ST_AsText function.
+                    expression = new FunctionCall(QualifiedName.of("ST_AsText"), ImmutableList.of(expression));
```

Comment:
Hi @HeidiHan0000 , are all Geometry values convertable to varchar, by ST_AsText? How about ST_AsBinary? Is it the case that either of the two works here?